src/Tuura/Concept/Circuit/Derived.hs

```diff
@@ -199,3 +201,14 @@ internals ints = interfaceConcept $ \s ->
                  if s `elem` ints
                  then Internal
                  else Unused
+
+combinationalGate :: Eq a => Expr a -> a -> CircuitConcept a
```

Comment:
Would `complexGate` not need to be `Eq a => Expr a -> Expr a -> a -> CircuitConcept`